### ISSUE TITLE
Falling back to VMA file mapping on NetBSD

### ISSUE THREAD

**Issue by fish4terrisa-MSDSM** [LekKit/RVVM#98]
I have tried rvvm on sdf.(The environment is netbsd 9.3)
It worked well ,but I got the warnings:
```
WARN: Falling back to VMA file mapping                                                                                               
WARN: Falling back to VMA file mapping                                                                                              
WARN: Falling back to VMA file mapping                                                                                               
WARN: Falling back to VMA file mapping
```
Is that a bug of RVVM or limition of netbsd?

**Comment by LekKit** [comment]
> Is that a bug of RVVM or limitation of netbsd?

It is one of implementations for allocating JIT executable memory, when all other methods have failed.

It is there to have basic JIT support even if some OS is very paranoid about allowing RVVM to be JITed via usual means.
Why it deserves a warning, this method might be too slow on some OSes (It was terribly slow on SerenityOS, for example) because it uses a file as a backend for JIT code.

Current methods of JIT heap allocation:
- RWX memory mapping (Simple, but disallowed on some OSes like OpenBSD, Linux hardened, and apparently NetBSD)
- Anonymous FD mapping (Works on most OSes that don't support previous one. Unimplemented on NetBSD yet.)
- Finally, creating a file in `/tmp` and mapping it an executable. This prints a warning because previous implementations are preferred.

Basically NetBSD might need some specific code to support it better, in `vma_ops.c` lib.

**Comment by LekKit** [comment]
Here are the implementations for other platforms:
https://github.com/LekKit/RVVM/blob/695cd2796b08ffd1bd661626513622255914e4e1/src/vma_ops.c#L92

There needs to be some syscall similar to `memfd_create()`, `shm_open(SHM_ANON)`, `shm_mkstemp()`. Not that I found any documentation on similar thing in NetBSD, yet.

Could you also please run `rvvm -v` in verbose mode and attach what is is saying before the warn?


UPD: Implementation in Mesa3D:
https://gitlab.freedesktop.org/mesa/mesa/-/blob/main/src/util/anon_file.c#L98

There doesn't seem any specific code for NetBSD. It is basically doing the same thing as I do with temporary files, altho it is storing them in `XDG_RUNTIME_DIR` which presumably is in `tmpfs`.
I will need to run a NetBSD installation myself to test it properly.

You may consider this warn harmless for now if you don't have severe perf drop comparing to RVVM on other platforms.

**Comment by LekKit** [comment]
Apparently this is SDF configuration quirk, not even a NetBSD issue (They disallow RWX `mmap()`, POSIX `shm_open()` and idk what else).
RVVM is simply too insistent on having a JIT that it goes all the way to use a file in `/tmp` for executable code, and succeeds (Even if it's not optimal). Maybe `/var/run` is a ram-backed directory more suitable for this.

**Comment by LekKit** [comment]
Silenced the warnings and improved safety/correctness in c6ed745. Reopen if there are additional issues with perf/stability.
